cadctap/cadctap/core.py

```diff
@@ -361,12 +361,37 @@ def _add_anon_option(parser):
         for g_action in m_group._group_actions:
             for o_string in g_action.option_strings:
                 if 'cert' in o_string or 'netrc' in o_string:
+                    # add the --anon authentication option
                     m_group.add_argument(
-                        '-a', action='store_true',
-                        help='use the service anonymously, short form')
-                    m_group.add_argument(
-                        '--anon', action='store_true',
-                        help='use the service anonymously, long form')
+                        '-a', '--anon', action='store_true',
+                        help='use the service anonymously')
+
```

Comment:
Please add a comment as to why it's so complicated: argparse does not work when option is added to the mutually exclusive group later after other arguments are added. This is a workaround to make it work.

Reply:
Done.